### ISSUE TITLE
[Bug]: KV cache allocation test fails on a Spyre host (no RuntimeContext)

### ISSUE THREAD

**Issue by jvlunteren** [torch-spyre/spyre-inference#766]
### Describe the bug

`test_kv_cache_shape_matches_runner_allocation` fails on a host with a Spyre card. It builds a runner and calls `initialize_kv_cache_tensors`, which transfers the KV pages with an explicit device_layout, and that transfer needs a `Spyre RuntimeContext`. A `layout-carrying .to(`) does not create one, and nothing in the test touches the device beforehand,

### How to reproduce

On a host with a Spyre card, at 0c72c0f:

`pytest tests/attention/test_spyre_attn.py::test_kv_cache_shape_matches_runner_allocation -m 'not upstream'`

Fails. Reproduced in a clean worktree at pristine upstream/main, so it is not dependent on local changes. Running device-touching tests first does not help.

### Additional context

Two separate problems, both worth fixing:

1. The test needs a device touch before the layout-carrying transfer, e.g. `torch.zeros(8, 8, dtype=torch.float16).to("spyre")`.
2. It needs a `spyre_available(`) skip. Tests needing real hardware guard themselves so that "all green" on a non-Spyre machine does not read as passing. Without the guard this test is silently inert on cardless runners.

Separately, `initialize_kv_cache_tensors` has no check that the allocated pages keep the slot axis whole at device position 0. The layout is correct today via slot_major_kv_layout, but that layout exists to work around torch-spyre #3705 rather than for the paged scatter, so a future layout change could regress the scatter to writing the wrong rows with nothing failing loudly.

### Checklist

- [x] I have searched for similar issues.
